Firestore/core/src/firebase/firestore/remote/remote_objc_bridge.mm

```diff
@@ -216,6 +218,53 @@ bool IsLoggingEnabled() {
   return [response description];
 }
 
+// DatastoreSerializer
+
+GCFSBatchGetDocumentsRequest* DatastoreSerializer::CreateLookupRequest(
+    const std::vector<DocumentKey>& keys) const {
+  GCFSBatchGetDocumentsRequest* request =
+      [GCFSBatchGetDocumentsRequest message];
+
+  request.database = [serializer_ encodedDatabaseID];
+  for (const DocumentKey& key : keys) {
+    [request.documentsArray addObject:[serializer_ encodedDocumentKey:key]];
+  }
+
+  return request;
+}
+
+grpc::ByteBuffer DatastoreSerializer::ToByteBuffer(
+    GCFSBatchGetDocumentsRequest* request) {
+  return ConvertToByteBuffer([request data]);
+}
+
+NSArray<FSTMaybeDocument*>* DatastoreSerializer::MergeLookupResponses(
+    const std::vector<grpc::ByteBuffer>& responses, Status* out_status) const {
+  std::map<DocumentKey, FSTMaybeDocument*> results;
```

Comment:
`std::map` was used in the [Objective-C implementation](https://github.com/firebase/firebase-ios-sdk/blob/master/Firestore/Source/Remote/FSTDatastore.mm#L265), I'm fine with sorting the `NSArray`.